Add process on failing release (#138)
meilisearch/integration-guides: guides/integration-release.md

> - At the top right part of the screen click on `Re-run all jobs`.

**This keeps consistency between the PRs history and the release changelogs**.

### What to do when Failing a Release 

> In some repositories, [github actions are triggered](https://github.com/meilisearch/meilisearch-js/blob/main/.github/workflows/publish.yml) after a release creates a tag. It might happen that one of these actions failed because pre-requisites are not met (i.e., the new version in the correct file). Don't worry; it is fixable! These are the steps: 

Reviewer: ```suggestion
In some repositories, [GitHub Actions are triggered](https://github.com/meilisearch/meilisearch-js/blob/main/.github/workflows/publish.yml) after a tag creation following the release publishment. It might happen that one of these actions failed because pre-requisites are not met (i.e., the new version in the correct file). Don't worry; it is fixable! These are the steps: 
```

Author: [Publishement is an archaic](https://www.merriam-webster.com/dictionary/publishment#:~:text=archaic,announcement%20of%20banns%20of%20marriage) term that should be replaced with publication, is it oke for you if I change it with `publication`?

Reviewer: Perfect! :)

Author: ```suggestion
In some repositories, [GitHub actions are triggered](https://github.com/meilisearch/meilisearch-js/blob/main/.github/workflows/publish.yml) after a tag creation following the release publication. It might happen that one of these actions failed because pre-requisites are not met (i.e., the new version in the correct file). Don't worry; it is fixable! These are the steps: 
```

> In some repositories, [GitHub actions are triggered](https://github.com/meilisearch/meilisearch-js/blob/main/.github/workflows/publish.yml) after a tag creation following the release publication. It might happen that one of these actions failed because pre-requisites are not met (i.e., the new version in the correct file). Don't worry; it is fixable! These are the steps: 

Reviewer: ```suggestion
In some repositories, [GitHub Actions are triggered](https://github.com/meilisearch/meilisearch-js/blob/main/.github/workflows/publish.yml) after a tag creation following the release publication. It might happen that one of these actions failed because pre-requisites are not met (i.e., the new version in the correct file). Don't worry; it is fixable! These are the steps: 
```

> 1. Delete the release: go to release, then click on the latest release and then on delete in the upper right corner.

Reviewer: ```suggestion
1. Delete the release: go to "Release" section on GitHub, then click on the latest release and then on "Delete" in the upper right corner.
```

> 1. Delete the release: go to "Release" section on GitHub, then click on the latest release and then on "Delete" in the upper right corner.

Reviewer: ```suggestion
1. Delete the release: go to the "Releases" section on GitHub, then click on the latest release and then on "Delete" in the upper right corner.
```

> 2. Delete the `tag`:  go to release, then click on `tags`, go to the latest tag, click on delete in the upper right corner.

Reviewer: ```suggestion
2. Delete the tag (`v0.X.Y`):  go to release, then click on the "Tag" section, go to the latest tag, click on "Delete" in the upper right corner.
```

> 2. Delete the tag (`v0.X.Y`):  go to release, then click on the "Tag" section, go to the latest tag, click on "Delete" in the upper right corner.

Reviewer: ```suggestion
2. Delete the tag (`v0.X.Y`):  go to the "Releases" section on GitHub, then click on the "Tags" section, go to the latest tag, click on "Delete" in the upper right corner.
```

3. Merge the PR with the missing pre-requisites (i.e., adding the current version in the correct file).
4. Once the merge completes, a new release draft is created. You can now publish it! 

> **For Example**

Reviewer: ```suggestion
**Failure situation example**
```

> I release `meilisearch-js@v0.2.0` but I forgot to update the version in my `package.json`. On release, a [`publish to npm`]((https://github.com/meilisearch/meilisearch-js/blob/main/.github/workflows/publish.yml)) GitHub action is triggered. A test is done to check if the version in `package.json` is the same as the `tag` created by the GitHub release. In this case, `v0.1.0` in my package.json and `v0.2.0` as a tag; thus, the GitHub action stops, and my package is not published on [npm](https://www.npmjs.com/).

Reviewer: ```suggestion
I release `meilisearch-js@v0.2.0` but I forgot to update the version in my `package.json`. On release, the [`publish to npm`]((https://github.com/meilisearch/meilisearch-js/blob/main/.github/workflows/publish.yml)) GitHub Action is triggered. A test is done to check if the version in `package.json` is the same as the `tag` created by the GitHub release. In this case, `v0.1.0` in my package.json and `v0.2.0` as a tag; thus, the GitHub Action stops, and my package is not published to [npm](https://www.npmjs.com/).
```
